server/protocols/jmap-draft-integration-testing/jmap-draft-integration-testing-common/src/test/java/org/apache/james/jmap/draft/methods/integration/FilterTest.java

```diff
@@ -240,6 +240,138 @@ public void setFilterShouldOverwritePreviouslyStoredRules() {
+                "      \"action\": {" +
+                "        \"appendIn\": {" +
+                "          \"mailboxIds\": [\"" + mailbox.serialize() + "\"]" +
+                "        }" +
+                "      }" +
+                "    }" +
+                "  ]}, " +
+                "\"#0\"" +
+                "]]")
+        .when()
+            .post("/jmap")
+        .then()
+            .body(ARGUMENTS + ".type", equalTo("invalidArguments"))
+            .body(ARGUMENTS + ".description", equalTo("`name` is mandatory"));
+    }
+
+    @Category(BasicFeature.class)
+    @Test
+    public void setFilterShouldRejectRuleWithInvalidRuleId() {
+        MailboxId mailbox = randomMailboxId();
+
+        given()
+            .header("Authorization", accessToken.asString())
+            .body("[[" +
+                "  \"setFilter\", " +
+                "  {" +
+                "    \"singleton\": [" +
+                "    {" +
+                "      \"id\": null," +
+                "      \"name\": \"some name\"," +
+                "      \"condition\": {" +
+                "        \"field\": \"subject\"," +
+                "        \"comparator\": \"contains\"," +
+                "        \"value\": \"question\"" +
+                "      }," +
+                "      \"action\": {" +
+                "        \"appendIn\": {" +
+                "          \"mailboxIds\": [\"" + mailbox.serialize() + "\"]" +
+                "        }" +
+                "      }" +
+                "    }" +
+                "  ]}, " +
+                "\"#0\"" +
+                "]]")
+        .when()
+            .post("/jmap")
+        .then()
+            .body(ARGUMENTS + ".type", equalTo("invalidArguments"))
+            .body(ARGUMENTS + ".description", equalTo("`id` is mandatory"));
+    }
+
+    @Category(BasicFeature.class)
+    @Test
+    public void setFilterShouldRejectRuleWithInvalidRuleCondition() {
+        MailboxId mailbox = randomMailboxId();
+
+        given()
+            .header("Authorization", accessToken.asString())
+            .body("[[" +
+                "  \"setFilter\", " +
+                "  {" +
+                "    \"singleton\": [" +
+                "    {" +
+                "      \"id\": \"3000-34e\"," +
+                "      \"name\": \"some name\"," +
+                "      \"condition\": null," +
+                "      \"action\": {" +
+                "        \"appendIn\": {" +
+                "          \"mailboxIds\": [\"" + mailbox.serialize() + "\"]" +
+                "        }" +
+                "      }" +
+                "    }" +
+                "  ]}, " +
+                "\"#0\"" +
+                "]]")
+        .when()
+            .post("/jmap")
+        .then()
+            .body(ARGUMENTS + ".type", equalTo("anErrorOccurred"))
+            .body(ARGUMENTS + ".description", equalTo("Failed to retrieve filter"));
+    }
+
+    @Category(BasicFeature.class)
```

Comment:
I don't think you have to put all these tests as BasicFeature, as a check on the memory implementation is sufficient for most of them.

Reply:
IMO tall of that just need to run on the memory implementation

Reply:
I agree with it.